noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/tests/previous_kernel_validator_builder/validate_proof.nr

```diff
@@ -0,0 +1,42 @@
+use crate::tests::previous_kernel_validator_builder::PreviousKernelValidatorBuilder;
+use types::constants::{
+    PRIVATE_KERNEL_INIT_INDEX, PRIVATE_KERNEL_INNER_INDEX, PRIVATE_KERNEL_RESET_INDEX,
+};
+
+impl PreviousKernelValidatorBuilder {
+    pub fn new_with_vk(vk_index: u32) -> Self {
+        let mut builder = Self::new();
+        let _ = builder.previous_kernel.in_vk_tree(vk_index);
+        builder
+    }
+}
+
+#[test]
+fn validate_proof__succeeds() {
+    let builder = PreviousKernelValidatorBuilder::new_with_vk(PRIVATE_KERNEL_INNER_INDEX);
+    builder.validate_proof([
+        PRIVATE_KERNEL_INIT_INDEX,
+        PRIVATE_KERNEL_INNER_INDEX,
+        PRIVATE_KERNEL_RESET_INDEX,
+    ]);
+}
+
+#[test(should_fail_with = "Invalid vk index")]
+fn validate_proof__vk_not_allowed_failed() {
+    let builder = PreviousKernelValidatorBuilder::new_with_vk(PRIVATE_KERNEL_INNER_INDEX);
+    builder.validate_proof([PRIVATE_KERNEL_INIT_INDEX, PRIVATE_KERNEL_RESET_INDEX]);
+}
+
+#[test(should_fail_with = "Invalid VK hash")]
+fn validate_proof__vk_hash_mismatch_fails() {
```

Comment:
Not for this PR - and more of a general nit because obviously this PR didn't introduce this! - but would be good to consistently use the same capitalisation of `vk` or `VK` everywhere. Extremely pedantic nit 😅